mesa/space.py

```diff
@@ -459,14 +459,19 @@ def move_agent_to_one_of(
             elif selection == "closest":
                 current_pos = agent.pos
                 # Find the closest position without sorting all positions
-                closest_pos = None
+                closest_pos = []
                 min_distance = float("inf")
                 for p in pos:
```

Comment:
```suggestion
                for p in agent.random.shuffle(pos):
```
I think with this single line change we could be done.